# down load the data 
cd /groups/nordborg/projects/cegs/alexandra/Public_data/He_et_al



wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/069/SRR14056769/SRR14056769_1.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/069/SRR14056769/SRR14056769_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/070/SRR14056770/SRR14056770_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/070/SRR14056770/SRR14056770_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/071/SRR14056771/SRR14056771_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/071/SRR14056771/SRR14056771_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/072/SRR14056772/SRR14056772_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/072/SRR14056772/SRR14056772_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/073/SRR14056773/SRR14056773_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/073/SRR14056773/SRR14056773_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/074/SRR14056774/SRR14056774_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/074/SRR14056774/SRR14056774_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/075/SRR14056775/SRR14056775_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/075/SRR14056775/SRR14056775_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/076/SRR14056776/SRR14056776_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/076/SRR14056776/SRR14056776_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/077/SRR14056777/SRR14056777_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/077/SRR14056777/SRR14056777_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/078/SRR14056778/SRR14056778_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/078/SRR14056778/SRR14056778_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/079/SRR14056779/SRR14056779_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/079/SRR14056779/SRR14056779_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/080/SRR14056780/SRR14056780_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/080/SRR14056780/SRR14056780_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/081/SRR14056781/SRR14056781_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/081/SRR14056781/SRR14056781_2.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/082/SRR14056782/SRR14056782_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/082/SRR14056782/SRR14056782_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/083/SRR14056783/SRR14056783_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/083/SRR14056783/SRR14056783_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/084/SRR14056784/SRR14056784_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/084/SRR14056784/SRR14056784_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/085/SRR14056785/SRR14056785_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/085/SRR14056785/SRR14056785_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/086/SRR14056786/SRR14056786_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/086/SRR14056786/SRR14056786_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/087/SRR14056787/SRR14056787_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR140/087/SRR14056787/SRR14056787_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/098/SRR16596898/SRR16596898_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/098/SRR16596898/SRR16596898_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/099/SRR16596899/SRR16596899_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/099/SRR16596899/SRR16596899_2.fastq.gz

wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/000/SRR16596900/SRR16596900_1.fastq.gz
wget -c -P . ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR165/000/SRR16596900/SRR16596900_2.fastq.gz


# all data - 150bp PE 
# Library construction (RiboMinus Plant Kit, Invitrogen; Ultra II Directional RNA Library Prep Kit, NEB)


samples=/groups/nordborg/user/aleksandra.kornienko/Documents/01_POSTDOC/03_Projects/Public_data/He_et_al_Methylation_free_Arabidopsis/He_et_al_srr_sample_name_RNAseq.txt
wc -l $samples 
#15 /groups/nordborg/user/aleksandra.kornienko/Documents/01_POSTDOC/03_Projects/Public_data/He_et_al_Methylation_free_Arabidopsis/He_et_al_srr_sample_name_RNAseq.txt


working_folder=/groups/nordborg/projects/cegs/alexandra/Public_data/He_et_al

export sample=`cat $samples | sed -n $numberofline,"$numberofline"p  | awk '{print $2}'`
export srr=`cat $samples | sed -n $numberofline,"$numberofline"p| awk '{print $1}'`

fastq1=$srr
fastq1+=_1.fastq.gz

fastq2=$srr
fastq2+=_2.fastq.gz

module load star/2.7.1a-foss-2018b

#align

mkdir BAM/$sample
export stargenomedir=/groups/nordborg/user/aleksandra.kornienko/analyses/STAR/STAR2.7.1a/STAR_TAIR10_noSJDB_sizeaccounted
export out=BAM/$sample/$sample.

#95% of Arabidopsis introns are smaller than 3.5kb

STAR --readFilesIn  $fastq1 $fastq2 --readFilesCommand zcat  \
--runThreadN 4 \
--alignIntronMax 6000 \
--alignMatesGapMax 6000 \
--genomeDir $stargenomedir \
--outFilterIntronMotifs RemoveNoncanonical \
--outFilterMismatchNoverReadLmax 0.1 \
--outFilterMismatchNmax 999 \
--outFilterMismatchNoverLmax 0.3 \
--outFilterMultimapNmax 10 \
--outReadsUnmapped None \
--alignSJoverhangMin 8 \
--outSAMattributes NH HI AS nM NM MD jM jI XS \
--outSAMtype BAM SortedByCoordinate  \
--runMode alignReads \
--twopassMode Basic \
--outFileNamePrefix $out



## create indexed bam file
module load samtools/1.9-foss-2018b
samtools index $working_folder/BAM/$sample/$sample.Aligned.sortedByCoord.out.bam  

#check BAM quality 
bam=$working_folder/BAM/$sample/$sample.Aligned.sortedByCoord.out.bam  

#check ribo and chloroplast contamination 
module load bedtools/2.27.1-foss-2018b

srun --mem-per-cpu=30G bamToBed -i $bam   | coverageBed -a /groups/nordborg/user/aleksandra.kornienko/analyses/Annotation/Athaliana_full_chromosomes_andRibo.bed.txt -b stdin > $working_folder/BAM/$sample/$sample.Rib_ChrC_contamination.txt
 
module unload  rseqc/2.6.5-foss-2018b-python-2.7.15

#make bigwig 
module load deeptools/3.1.2-foss-2018b-python-2.7.15


srun --mem-per-cpu=30G bamCoverage -b $bam  --filterRNAstrand forward -o $working_folder/BAM/$sample/$sample.F.bw --binSize=25

srun --mem-per-cpu=30G bamCoverage -b $bam  --filterRNAstrand reverse -o $working_folder/BAM/$sample/$sample.R.bw --binSize=25
 


 









